.github/workflows/ci.yml

```
@@ -109,12 +109,49 @@ jobs:
           # if no PR number provided, simply exit...
           [[ -z "$TARGET_PR" ]] && { echo "No PR Number provided, exiting..." ; exit 0; }
 
+          echo "PR Number provided... ${TARGET_REPOSITORY}:#${TARGET_PR}"
+
+          BODY_PAYLOAD=""
           # if all the previous steps finished successfully, create a comment on the PR with the "success" information
           if [ "$TEST_RESULT" == "success" ] && [ "$TEST_LINT_RESULT" == "success" ] && [ "$MODULE_LINT_RESULT" == "success" ]; then
             BODY_PAYLOAD="[Microsoft CI Bot] TL;DR; success :thumbsup:\n\nYou can check the status of the CI Pipeline logs here ; https://github.com/${CURRENT_REPOSITORY}/actions/runs/$RUN_ID"
-            curl --request POST --header "Authorization: token ${GITHUB_TOKEN}" --header "Accept: application/vnd.github.v3+json" https://api.github.com/repos/${TARGET_REPOSITORY}/issues/${TARGET_PR}/comments --data "{\"body\":\"${BODY_PAYLOAD}\"}"
           # if at least one of the previous steps failed, create a comment on the PR with the "failure" information
           elif [ "$TEST_RESULT" == "failure" ] || [ "$TEST_LINT_RESULT" == "failure" ] || [ "$MODULE_LINT_RESULT" == "failure" ]; then
             BODY_PAYLOAD="[Microsoft CI Bot] TL;DR; failure :facepalm:\n\nYou can check the status of the CI Pipeline logs here ; https://github.com/${CURRENT_REPOSITORY}/actions/runs/$RUN_ID"
-            curl --request POST --header "Authorization: token ${GITHUB_TOKEN}" --header "Accept: application/vnd.github.v3+json" https://api.github.com/repos/${TARGET_REPOSITORY}/issues/${TARGET_PR}/comments --data "{\"body\":\"${BODY_PAYLOAD}\"}"
+          fi
+
+          echo "Comment message is ready..."
+          echo "${BODY_PAYLOAD}"
+
+          # if pipeline has something to report back to the PR
+
+          if [ -z "$BODY_PAYLOAD" ]
```

We follow [the google style guide for shell](https://google.github.io/styleguide/shellguide.html), and one of the recommendations is to prefer `[[]]` over `[]` (see [here](https://google.github.io/styleguide/shellguide.html#s6.3-tests) for explanation).

```suggestion
          if [[ -z "$BODY_PAYLOAD" ]]
```